src/functions/reportGen.ts

```diff
@@ -16,22 +23,38 @@ const reportGen: Handler = async (event: any, context?: Context, callback?: Call
         console.error("ERROR: event is not defined.");
         return;
     }
-
     const s3BucketService: S3BucketService = Injector.resolve<S3BucketService>(S3BucketService);
     const reportService: ReportGenerationService = Injector.resolve<ReportGenerationService>(ReportGenerationService);
     const retroUploadPromises: Array<Promise<ManagedUpload.SendData>> = [];
+    const testStationsService: TestStationsService = Injector.resolve<TestStationsService>(TestStationsService);
+    const notifyClient = new NotifyClient(Configuration.getInstance().getGovNotifyConfig().api_key);
+    const notifyService: NotificationService = new NotificationService(notifyClient);
+    const notificationData: NotificationData = new NotificationData();
+    let testResultsList: any;
 
     event.Records.forEach((record: any) => {
         const visit: any = JSON.parse(record.body);
         const retroUploadPromise = reportService.generateATFReport(visit)
-        .then((generationServiceResponse: { fileName: string, fileBuffer: Buffer}) => {
-
+        .then((generationServiceResponse: { fileName: string, fileBuffer: Buffer, testResults: any}) => {
```

Comment:
@acozianu @oalexandru @KHodderDVSA @bsiddappa  @akothiyal-deloitte I think we should aim to keep the function as simple as possible and do all the logic in the services, but I am happy to hear other suggestions. I guess that might imply that one service should call another.

Reply:
Agree with Adrian, functions should be kept as light as possible. I see them more as kind of controllers with actual logic being done by services

Reply:
For me this sounds more like a refactoring ticket because this is something that wasn't change in this PR.
This part of the logic was created by George long time ago.